### ISSUE TITLE
Lint idea: drop bounds

### ISSUE THREAD

**Issue by notriddle** [rust-lang/rust-clippy#3773]
Code in this form should trigger a warning: bounds by drop does not accomplish anything, since t.drop() cannot be directly called, and non-Drop types may have drop glue anyway.

```rust
fn foo<T: Drop>(t: T) { ... }
```

**Comment by CodesInChaos** [comment]
Same for:
```
trait MyTrait: Drop {}
```

**Comment by notriddle** [comment]
Cool, I've started implementing it then!
